alot/commands/envelope.py

```diff
@@ -570,3 +570,59 @@ def apply(self, ui):
             envelope.encrypt_keys = {}
         # reload buffer
         ui.current_buffer.rebuild()
+
+
+@registerCommand(
+    MODE, 'tag', forced={'action': 'add'},
+    arguments=[(['tags'], {'help': 'comma separated list of tags'})],
+    help='add tags to message',
+)
+@registerCommand(
+    MODE, 'retag', forced={'action': 'set'},
+    arguments=[(['tags'], {'help': 'comma separated list of tags'})],
+    help='set message tags.',
+)
+@registerCommand(
+    MODE, 'untag', forced={'action': 'remove'},
+    arguments=[(['tags'], {'help': 'comma separated list of tags'})],
+    help='remove tags from message',
+)
+@registerCommand(
+    MODE, 'toggletags', forced={'action': 'toggle'},
+    arguments=[(['tags'], {'help': 'comma separated list of tags'})],
+    help='flip presence of tags on message',
+)
+class TagCommand(Command):
+
+    """manipulate message tags"""
+    repeatable = True
+
+    def __init__(self, tags=u'', action='add', **kwargs):
+        """
+        :param tags: comma separated list of tagstrings to set
+        :type tags: str
```

Comment:
The type is listed here as a `str` but the default argument is `unicode`. Let's either set this to `basestring` or `unicode` and use an `assert` to ensure it is a unicode object.